server/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -178,7 +178,7 @@ public Query buildShapeQuery(SearchExecutionContext context, MappedFieldType fie
             );
         }
         final GeoShapeQueryable ft = (GeoShapeQueryable) fieldType;
-        return new ConstantScoreQuery(ft.geoShapeQuery(shape, fieldType.name(), strategy, relation, context));
+        return new ConstantScoreQuery(ft.geoShapeQuery(context, fieldType.name(), strategy, relation, shape));
```

Comment:
```suggestion
        return new ConstantScoreQuery(ft.geoShapeQuery(context, fieldType.name(), relation, shape));
```

Reply:
Needed for compatibility with `LegacyGeoShapeFieldMapper`